### PR TITLE
chore: Support Cython 3.x

### PR DESCRIPTION
**Pull request by jetuk**
Several updates required to handle building with Cython 3.x.

- Include `noexcept` for GLPK callback functions.
- Handle name mangling of attributes using __. Cython 3 aligns more closely with Python 3 in this respect.

There should be a review of the use of `except` given the new behaviour in Cython 3, but that is not done here.

### CODE REVIEW

**Comment by jetuk** [comment]
This fixes #1102

**Comment by jetuk** [comment]
It might also be possible to remove the Numpy API warnings, but that could be a separate PR.

File: pyproject.toml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 [build-system]
-requires = ["setuptools>=42", "wheel", "setuptools_scm[toml]>=3.4", "cython<3", "numpy"]
```
Comment (Batch21) [inline pyproject.toml:2]: Changes to attribute names mean that Pywr can't now be built with cython<3 so this should probably be updated to "cython>=3"

Comment (jetuk) [reply pyproject.toml:2]: Done

**Review by Batch21 — APPROVED** [review]

**Pull request merged**